components/engine_traits/src/engine.rs

```diff
@@ -61,11 +61,24 @@ pub trait KvEngine:
     fn bad_downcast<T: 'static>(&self) -> &T;
 }
 
+/// TabletAccessor is the trait to access all the tablets with provided accessor
+///
+/// For single rocksdb instance, it essentially accesses the global kvdb with the accessor
+/// For multi rocksdb instances, it accesses all the tablets with the accessor
+pub trait TabletAccessor<EK> {
+    /// Loop visit all opened tablets by the specified function.
+    fn for_each_opened_tablet(&self, _f: Box<dyn FnMut(u64, u64, &EK) + '_>);
```

Comment:
How about `impl FnMut(u64, u64, &EK)`?

Reply:
no, &EK is from the callee, not the caller. But the FnMut is from the caller's behavior and it can be anything.